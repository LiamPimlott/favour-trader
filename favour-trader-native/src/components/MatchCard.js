import React, {Component} from 'react';
import {StyleSheet, View, Text, } from 'react-native';
import { Card, Avatar, Button, Icon} from 'react-native-elements'
import ReviewSkillsModal from "./ReviewSkillsModal";
class MatchCard extends React.Component{
    state = {
        modalVisible: false,
    };

    setModalVisible() {
        this.setState({modalVisible: !this.state.modalVisible});
    }

    render() {
        const { match } = this.props;
        return (
            <View key={match._id}>
                <Card
                    title={match.name.first + ' ' + match.name.last}>
                    <View style={styles.avatar}>
                        <Avatar
                            large
                            rounded
                            source={{uri: "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg"}}
                            activeOpacity={0.7}
                        />
                    </View>

                    <Text style={styles.email} >
                        {match.email}
                    </Text>
                    <Button
                        icon={<Icon name={'account-circle'} color={'white'} />}
                        backgroundColor='#03A9F4'
                        buttonStyle={styles.button}
                        title='View Profile' 
                        onPress = {this.props.passUserID.bind(this,match._id)}/>
                    <Button
                        icon={<Icon name={'info'} color={'white'} />}
                        backgroundColor='#03A9F4'
                        buttonStyle={styles.button}
                        title='Review Skills'
                        onPress={() => {
                            this.setModalVisible(!this.state.modalVisible, match.has, match.wants);
                        }}
                    />
                </Card>
                {
                    (this.state.modalVisible) ? (
                        <ReviewSkillsModal toggle={this.setModalVisible.bind(this)} isVisible={this.state.modalVisible} has={match.has} wants={match.wants}/>
                    ) : (<View />)
                }
            </View>
        )
    }
}

const styles = StyleSheet.create({
    email: {
        fontSize: 20,
        marginBottom: 10,
        textAlign: 'center',
    },
    avatar: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    button: {
        borderRadius: 0,
        marginLeft: 0,
        marginRight: 0,
        marginBottom: 10,
        width: 300
    }
});

export default MatchCard;