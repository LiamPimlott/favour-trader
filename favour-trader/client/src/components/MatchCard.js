import React, { Component } from 'react';
import { Card, Button, } from 'antd';
import { Link } from 'react-router-dom';
import Avatar from 'react-avatar';

class MatchCard extends Component {
    render() {
        const {user, reveal} = this.props;
        return (
            <Card title={user.name.first + " " + user.name.last.charAt(0) + "."} style={{ width: 300, marginTop: '10px' }}>
                <Avatar style={{marginBottom: '20px'}} src="http://lorempixel.com/400/200/" size={125} round={true} />
                <a className="active" style={ {position: 'absolute', top: '50%', color:'blue'} } onClick={reveal}>Review Skills</a>
                <p style={{margin: '20px', height: '70px', width: '210px', fontSize: '13px'}}>
                        {
                            (user.about) ?
                                ( 
                                    (user.about.length > 100) ?
                                        (user.about.substring(0, 99) + " ...Read more on Profile") : 
                                        (user.about)
                                ) :
                                ("Check out my Profile!")
                        }
                </p>
                <Link to={'/Profile/'+user._id}>
                    <Button type="primary" title={'View Profile'} style={{marginTop: '0px'}}>View Profile</Button>
                </Link>
            </Card>
        );
    }
}

export default MatchCard;