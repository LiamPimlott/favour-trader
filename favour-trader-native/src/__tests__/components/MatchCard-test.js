import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import moxios from "moxios";
import MatchCard from '../../components/MatchCard.js';
import toJson from 'enzyme-to-json'; //added this line

beforeEach(function () {
    const Adapter = require("enzyme-adapter-react-16");
    configure({adapter: new Adapter()});
    moxios.install();
});

describe('Testing MatchCard component', () => {
    it('renders as expected', () => {
        const match ={
            _id : 'testId',
            name : { fisrt : '',
                    last  : ''},
            email : '',
            has : '',
            wants : ''
        }
        const bind = {bind:jest.fn()};
        const userId='';
        const passUserID={bind:jest.fn()};
        const wrapper = shallow(
            <MatchCard match={match} bind={bind} userId={userId} passUserID={passUserID}/>
        );
        expect(toJson(wrapper)).toMatchSnapshot(); //edited this line
    });
});