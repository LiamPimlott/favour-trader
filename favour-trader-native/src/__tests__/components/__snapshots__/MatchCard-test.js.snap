// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MatchCard component renders as expected 1`] = `
<View
  key="testId"
>
  <Card
    title="undefined "
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Avatar
        activeOpacity={0.7}
        editButton={
          Object {
            "iconColor": "#fff",
            "iconName": "mode-edit",
            "iconType": "material",
            "size": null,
            "style": null,
            "underlayColor": "#000",
          }
        }
        large={true}
        onEditPress={null}
        rounded={true}
        showEditButton={false}
        source={
          Object {
            "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg",
          }
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    />
    <Button
      TouchableComponent={[Function]}
      backgroundColor="#03A9F4"
      buttonStyle={
        Object {
          "borderRadius": 0,
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "width": 300,
        }
      }
      clear={false}
      disabled={false}
      icon={
        <Icon
          color="white"
          name="account-circle"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      title="View Profile"
    />
    <Button
      TouchableComponent={[Function]}
      backgroundColor="#03A9F4"
      buttonStyle={
        Object {
          "borderRadius": 0,
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "width": 300,
        }
      }
      clear={false}
      disabled={false}
      icon={
        <Icon
          color="white"
          name="info"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      title="Review Skills"
    />
  </Card>
  <View />
</View>
`;
